src/main/java/org/spongepowered/api/item/inventory/ItemStackComparators.java

```diff
@@ -50,7 +44,7 @@
      * ItemStack.equals(ItemStack) for ItemStacks with extra attached data,
      * different damage values, or different sizes.
      */
-    public static final Comparator<ItemStack> TYPE = new Type();
+    public static final Comparator<ItemStack> TYPE = DummyObjectProvider.createFor(Comparator.class, "TYPE");
```

Comment:
`DummyObjectProvider.createExtendedFor` for all of these should make it so you don't need your unchecked suppression.